src/genotyper.cc

```diff
@@ -347,14 +337,105 @@ static Status translate_genotypes(const genotyper_config& cfg, const unified_sit
+/// min_ref_depth tracks the minimum depth of ref-records for a given sample,
+/// and will be set to -1 if there are no ref records processed for a sample
+/// min_ref_depth is expected to be initialized to -1 for all samples.
+/// Modifications to genotypes vector is explained below
+/// The representative record is expected to be passed into translate_genotypes for genotyping
+/// =====================================================
+/// Pre conditions:
+/// min_ref_depth should be initialized to -1 for all samples
+/// ans should be empty when passed in
+/// ======================================================
+/// Expected behaviors:
+/// No records given
+///      ans returned as empty, no modification to genotypes and failed_ref_depth vectors
+/// Records do not span the entire range of site
+///      ans returned as empty, min_ref_depth updated accordingly, no change to genotypes
+/// Records span entire range, and consist of all gvcf records
+///      ans set as the first gvcf record passed, min_ref_depth updated accordingly, no change to genotypes
+/// Records span entire range, and consist of exactly 1 vcf variant record
+///      ans set as the 1 vcf variant record, min_ref_depth updated accordingly, no change to genotypes
+/// Records span entire range, and consists of >1 vcf variant record
+///      ans returned as empty, min_ref_depth updated accordingly, genotypes for all samples in this dataset are updated to be lost calls with RNC=LostAllele
+
+Status find_rep_record(const genotyper_config& cfg, const unified_site& site,
+                       const vector<shared_ptr<bcf1_t>>& records, const string& dataset,
+                       const bcf_hdr_t* hdr, int bcf_nsamples,
+                       const map<int, int>& sample_mapping, vector<one_call>& genotypes,
+                       vector<int>& min_ref_depth, shared_ptr<bcf1_t>& ans) {
+
+
+    if (ans) {
+        return Status::Invalid("find_rep_record: ans is not empty when passed into the function.");
+    }
+    // Clear ans to be empty
+    ans.reset();
+
+    Status s;
+    unique_ptr<AlleleDepthHelper> depth;
+
+    // To track the total range covered by the records
+    vector<range> record_rngs;
+    for (auto& record: records) {
+        record_rngs.push_back(range(record.get()));
+        S(AlleleDepthHelper::Open(cfg, dataset, hdr, record.get(), depth));
+
+        // Update min_ref_depth vector
+        for (int i=0; i<bcf_nsamples; i++) {
+            // TODO: Consider support for hom ref records
+            int ref_depth = depth->get_gvcf_depth(i);
+            int mapped_sample = sample_mapping.at(i);
+            assert(mapped_sample < min_ref_depth.size());
+            if (min_ref_depth[mapped_sample] < 0) {
+                min_ref_depth[mapped_sample] = ref_depth;
+            } else {
+                min_ref_depth[mapped_sample] = min(min_ref_depth[mapped_sample], ref_depth);
+            }
```

Comment:
I think there might be a bug here in the case where we see a reference confidence record, followed by a variant record. When we get to the variant record, `ref_depth` be set to -1 by `get_gvcf_depth`, and we'll use that to overwrite an existing value of `min_ref_depth[mapped_sample]` from the previous reference confidence record. It looks like the join_records_insufficient_ref_depth.yml doesn't exercise this particular case.